build.gradle.kts

```diff
@@ -106,3 +106,52 @@ if (gradle.startParameter.taskNames.contains("listTestsInPartition")) {
     }
   }
 }
+
+if (gradle.startParameter.taskNames.contains("generateFossaConfiguration")) {
+  tasks {
+    val generateFossaConfiguration by registering {
+      group = "Help"
+      description = "Write .fossa.yml config file"
+
+      doLast {
+        File(".fossa.yml").printWriter().use { writer ->
+          writer.println("version: 3")
+          writer.println()
+          writer.println("targets:")
+          writer.println("  only:")
+          writer.println("    # only scanning the modules which are published")
+          writer.println("    # (as opposed to internal testing modules")
+          rootProject.subprojects
+            .sortedBy { it.findProperty("archivesName") as String? }
+            .filter { !it.name.startsWith("bom") }
+            .filter { it.plugins.hasPlugin("maven-publish") }
+            .forEach {
+              writer.println("    - type: gradle")
+              writer.println("      path: ./")
+              writer.println("      target: '${it.path}'")
+            }
+          writer.println()
+          writer.println("experimental:")
+          writer.println("  gradle:")
+          writer.println("    configurations-only:")
+          writer.println("      # consumer will only be exposed to these dependencies")
+          writer.println("      - runtimeClasspath")
+        }
+      }
+
+      // disable all tasks to stop build
+      subprojects {
+        tasks.configureEach {
+          enabled = false
+        }
+      }
+    }
+  }
+
+  // disable all tasks to stop build
+  project.tasks.configureEach {
+    if (this.name != "generateFossaConfiguration") {
+      enabled = false
+    }
+  }
```

Comment:
I'm not sure you need this. `listTestsInPartition` uses this because it is run with `./gradlew check -x spotlessCheck listTestsInPartition` We want to get names of all tests tasks that `check` would trigger but we don't want to actually run them.